packages/twenty-server/src/modules/workflow/workflow-executor/workflow-actions/delay/delay.workflow-action.ts

```diff
@@ -0,0 +1,77 @@
+import { Injectable } from '@nestjs/common';
+
+import { type WorkflowAction } from 'src/modules/workflow/workflow-executor/interfaces/workflow-action.interface';
+
+import { InjectMessageQueue } from 'src/engine/core-modules/message-queue/decorators/message-queue.decorator';
+import { MessageQueue } from 'src/engine/core-modules/message-queue/message-queue.constants';
+import { MessageQueueService } from 'src/engine/core-modules/message-queue/services/message-queue.service';
+import {
+  WorkflowStepExecutorException,
+  WorkflowStepExecutorExceptionCode,
+} from 'src/modules/workflow/workflow-executor/exceptions/workflow-step-executor.exception';
+import { type WorkflowActionInput } from 'src/modules/workflow/workflow-executor/types/workflow-action-input';
+import { type WorkflowActionOutput } from 'src/modules/workflow/workflow-executor/types/workflow-action-output.type';
+import { findStepOrThrow } from 'src/modules/workflow/workflow-executor/utils/find-step-or-throw.util';
+import { isWorkflowDelayAction } from 'src/modules/workflow/workflow-executor/workflow-actions/delay/guards/is-workflow-delay-action.guard';
+import { type WorkflowDelayActionInput } from 'src/modules/workflow/workflow-executor/workflow-actions/delay/types/workflow-delay-action-input.type';
+
+export type DelayedResumeJobData = {
+  workspaceId: string;
+  workflowRunId: string;
+  stepId: string;
+};
+
+@Injectable()
+export class DelayWorkflowAction implements WorkflowAction {
+  constructor(
+    @InjectMessageQueue(MessageQueue.workflowQueue)
+    private readonly messageQueueService: MessageQueueService,
+  ) {}
+
+  async execute({
+    currentStepId,
+    steps,
+    runInfo,
+  }: WorkflowActionInput): Promise<WorkflowActionOutput> {
+    const step = findStepOrThrow({
+      stepId: currentStepId,
+      steps,
+    });
+
+    if (!isWorkflowDelayAction(step)) {
+      throw new WorkflowStepExecutorException(
+        'Step is not a delay action',
+        WorkflowStepExecutorExceptionCode.INVALID_STEP_TYPE,
+      );
+    }
+
+    const settings = step.settings.input as WorkflowDelayActionInput;
+
+    if (!settings.scheduledDateTime) {
+      throw new WorkflowStepExecutorException(
+        'Scheduled date time is required for delay action',
+        WorkflowStepExecutorExceptionCode.INVALID_STEP_TYPE,
+      );
+    }
+
+    const scheduledDate = new Date(settings.scheduledDateTime);
+    const now = new Date();
+    const delayInMs = scheduledDate.getTime() - now.getTime();
+
+    await this.messageQueueService.add<DelayedResumeJobData>(
+      'DelayedResumeJob',
```

Comment:
If the queue is full then the delay might not be respected so we might want to do a higher priority queue for those jobs (MessageQueue enum: add workflowDelayedJobsQueue). We don't use priorities right now but a good opportunity to start introducing them. Let's also put the emailQueue in priority / do something clean. lower numbers = higher priority in BullMQ.